automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/notebooks/WorkspaceNotebooksPage.scala

```diff
@@ -0,0 +1,207 @@
+  def withNewCluster(clusterName: String,
+                     masterMachineType: String = "n1-standard-4",
+                     masterDiskSize: Int = 500,
+                     workers: Int = 0,
+                     workerMachineType: String = "n1-standard-4",
+                     workerDiskSize: Int = 500,
+                     workerLocalSSDs: Int = 0,
+                     preemptibleWorkers: Int = 0,
+                     extensionURI: Option[GcsPath] = None,
+                     customScriptURI: Option[GcsPath] = None,
+                     labels: Map[String, String] = Map())(testCode: (WorkspaceNotebooksPage) => Any) = {
+
+    openCreateClusterModal.createCluster(clusterName, masterMachineType, masterDiskSize, workers, workerMachineType, workerDiskSize, workerLocalSSDs, preemptibleWorkers, extensionURI, customScriptURI, labels)
+    await text clusterName
+    assert(getClusterStatus(clusterName) == "Creating")
+    logger.info("Creating dataproc cluster " + clusterName)
+    waitUntilClusterIsRunning(clusterName)
+    logger.info("Created dataproc cluster " + clusterName)
+
+    testCode
+
+    openDeleteClusterModal(clusterName).deleteCluster
+    await condition (getClusterStatus(clusterName) == "Deleting")
+    logger.info("Deleting dataproc cluster " + clusterName)
+    waitUntilClusterIsDeleted(clusterName)
+    await text noClustersMessage
+    logger.info("Deleted dataproc cluster " + clusterName)
+  }
+
+  // Goes to a Jupyter page. Currently doesn't return a page object
+  // because Jupyter functionality is tested in leo automated tests.
+  // Just check that it loads
+  def openJupyterPage(clusterName: String): JupyterPage = {
+    val initialWindowHandles = windowHandles
+
+    val clusterLink = Link(clusterName + "-link")
+    await ready clusterLink
+    clusterLink.doClick()
+
+    await condition (windowHandles.size > 1)
+    val jupyterWindowHandle = (windowHandles -- initialWindowHandles).head
+
+    switch to window(jupyterWindowHandle)
+    new JupyterPage().awaitLoaded()
+  }
+
+  private def waitUntilClusterIsRunning(clusterName: String) = {
+    while (getClusterStatus(clusterName) == "Creating") { Thread sleep 10000 }
+  }
+
+  private def waitUntilClusterIsDeleted(clusterName: String) = {
+    while (getClusterStatus(clusterName) == "Deleting") { Thread sleep 10000 }
+  }
+
+  def openCreateClusterModal(): CreateClusterModal = {
+    openCreateClusterModalButton.isStateEnabled
+    openCreateClusterModalButton.doClick()
+    await ready new CreateClusterModal
+  }
+
+  def openDeleteClusterModal(clusterName: String): DeleteClusterModal = {
+    val openDeleteClusterModalButton: Button = Button(clusterName + "-delete-button")
+
+    openDeleteClusterModalButton.doClick()
+    await ready new DeleteClusterModal(clusterName)
+  }
+
+  def getClusterStatus(clusterName: String): String = {
+    val clusterStatusId = testId(clusterName + "-status")
+    readText(clusterStatusId)
+  }
```

Comment:
Possible status: `Unknown, Creating, Running, Updating, Error, Deleting, Deleted`
Would be good to end test when inside `waitUntilClusterIsRunning` and `waitUntilClusterIsDeleted`,  status == `Error` or `Unknown`?